### PR TITLE
docs: add Angular CLI example

### LINKED ISSUE #170: Add Angular
## What is the framework/ technology you want to integrate Jest Preview to?
Angular

## Useful links
- https://angular.io/
- https://testing-library.com/docs/angular-testing-library/intro

## Do you want to add this example on your own?
I need help.

### PR DESCRIPTION
**Pull request by LayZeeDK**
## Summary/ Motivation (TLDR;)
Add an Angular CLI example to the repository.

## Related issues
- Closes #170

@nvh95 @ntt261298 
Here's an example using Angular CLI with Jest via [@angular-builders/jest](https://www.npmjs.com/package/@angular-builders/jest).

### CODE REVIEW

**Review by nvh95 — CHANGES_REQUESTED** [review]
Thank you @LayZeeDK very much for your wonderful PR. I hope that there are more Angular folks will use `jest-preview` to save their time after your PR.

I left some feedback. I can brief those:
- [ ] Rename current README to `README-angular.md` then add `README.md` to instruct integrating `jest-preview` to an Angular project.
- [ ] Update `jest.config.js` to
  - [ ] Better Intellisense
  - [ ] Add CSS processor
  - [ ] Add file processor
  - [ ] Only watch file in `src` (by adding `roots`)
- [ ] Add test watch mode to `package.json`
- [ ] Include "counter" when `ng serve`
- [ ] Can you add a sample of using CSS and Sass?
- [ ] Can you add an image using URL (beside base64)?
- [ ] import CSS file directly in `src/test-setup.ts` to support styling. `jest-preview/transforms/css` will handle it for you

Again. Thank you very much. Please let me know if you want to discuss anything regarding my feedback. Thanks.

**Comment by nvh95** [comment]
@LayZeeDK 
> I just read the raw global CSS stylesheet and append it to the document. I don't know how we would add Sass support although it is supported by Angular CLI.

I see, I left a comment in the feedback above. You can use CSS file transform, then import `.css` and `.scss` file directly. More at https://www.jest-preview.com/docs/getting-started/installation/#2-configure-jests-transform-to-transform-css-and-files

> Additionally, it would be interesting to run Angular in development mode to enable Angular Developer Tools.

Sorry, I haven't got your point. (Please forgive me I haven't had a chance to use Angular in production yet)

> I plan to add an Angular Testing Library example even though Angular's built-in TestBed API seems to work well.

That would be awesome. Currently, all existing examples are using `@testing-library` packages. But Jest Preview is framework agnostic, so it just needs a JSDOM environment. Thank you very much for this plan.

File: examples/angular/src/test-setup.ts
Comment (LayZeeDK) [reply examples/angular/src/test-setup.ts]: This is great, I have added import statements to `test-setup.ts` matching the global styles that are loaded via `angular.json`.

Comment (LayZeeDK) [reply examples/angular/src/test-setup.ts]: In Angular, styles are added through the `styles` and `styleUrls` options for the `Component` decorator. The `jest-preset-angular` removes styles but transforms templates for components using `templateUrl` , see https://github.com/thymikee/jest-preset-angular/blob/main/src/transformers/replace-resources.ts#L14-L44In Angular, styles are added through the `styles` and `styleUrls` options for the `Component` decorator. The `jest-preset-angular` removes styles but transforms templates for components using `templateUrl` to `require` or `import` statements, see https://github.com/thymikee/jest-preset-angular/blob/main/src/transformers/replace-resources.ts#L14-L44

**Comment by LayZeeDK** [comment]
>> Additionally, it would be interesting to run Angular in development mode to enable Angular Developer Tools.

> Sorry, I haven't got your point. (Please forgive me I haven't had a chance to use Angular in production yet)

Angular has two runtime modes:
- Production mode, used for deployed apps
- Development mode, used for local app development

The `ng serve` command Angular developers use to start a Webpack Development Server usually runs Angular in development mode which adds runtime checks that degrade performance and enables Angular Developer Tools to be used as a web extension.

I'm not sure how to instruct the `@angular-builders/jest:run` Angular CLI builder to start the application in development or whether this is something Jest Preview could support.

File: examples/angular/src/test-setup.ts
Comment (LayZeeDK) [reply examples/angular/src/test-setup.ts]: Please have another look at your convenience, @nvh95

Comment (nvh95) [reply examples/angular/src/test-setup.ts]: I have created a new issue to track this at #237. If you have any ideas, please let me know there. Thanks.

**TLDR:** 
- To fully support CSS, we need to make changes to Angular jest code transformation (in this case: `jest-preset-angular`)
- Now, we can work around by `import`ing CSS files directly.
- No workaround for `styles` yet.

**Review by nvh95 — APPROVED** [review]
Hi @LayZeeDK. Thank you for your updates. You resolved all the suggestions I made before. I ran the example and I think it's good for now.
Even though the styles are not fully processed. We can work around by #237 until `jest-preview` can support it natively.

Your PR looks good to me. Please mark the PR as ready, I will merge it right after this. I can do the workaround with `styleUrls` after merging this PR (You are welcome if you want to import those CSS files directly).

Again, thank you very much for your help.

**Comment by LayZeeDK** [comment]
@nvh95
I used the workaround to import component styles in tests and documented this in the example project's readme file.

**Comment by nvh95** [comment]
@LayZeeDK 🚀🚀
@all-contributors please add @LayZeeDK for example

**Pull request merged**
